### DISCUSSION TITLE
cx_Freeze.exception.OptionError: no base named 'console'

### DISCUSSION THREAD

**Discussion by angelotrivelli** [marcelotduarte/cx_Freeze#2384] [Q&A]
I've been trying to use cx-freeze to create an msi installer for a windows CLI application. It had worked a few months ago, but now I can't get it to work. I am reaching out for some clues on how to proceed. The problem is that it won't build. 

To troubleshoot, I've removed all dependencies and made an application that just prints "hello world". The problem persists. 

Below is what happens when I try to build. It complains about "no base named 'console' ('console-cpython-311-win_amd64')"...

```
PS C:\projects\vscode\example> python .\setup.py build
Traceback (most recent call last):
  File "C:\projects\vscode\example\setup.py", line 23, in <module>
    the_application = [Executable(script        = "app.py",
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\projects\vscode\example\.venv\Lib\site-packages\cx_Freeze\executable.py", line 47, in __init__
    self.base = base
    ^^^^^^^^^
  File "C:\projects\vscode\example\.venv\Lib\site-packages\cx_Freeze\executable.py", line 87, in base
    raise OptionError(msg)
cx_Freeze.exception.OptionError: no base named 'console' ('console-cpython-311-win_amd64')
```

My [setup.py file](https://github.com/angelotrivelli/example/blob/main/setup.py) is very simple (see below). I specified `base = None` because this is a CLI application. The fact that I stripped this down to no dependencies and still have a problem suggests that I am missing something very fundamental. Is my setup.py file OK?

```
from cx_Freeze import Executable, setup

exe_options = {
    "include_files": ["app_icon.ico"],
    "packages": []
}

msi_options = {
    'upgrade_code': '{dddf903f-a781-44ba-aded-cc10b711495d}',
    'add_to_path': True
}

# this is for a cli app, so we don't need a base?
base = None
    
the_application = [Executable(script        = "app.py",
                              base          = base,
                              shortcut_name = "example [1.0.0]",
                              shortcut_dir  = "DesktopFolder",
                              icon          = "app_icon.ico")]

setup(
    name = "my_example",
    version = "1.0.0",
    description = "exercise cx-freeze setup script for building a python app with msi installer",
    executables = the_application,
    options = {"build_exe": exe_options,
               "bdist_msi": msi_options},
)
```

I should add that I use poetry. I am on a windows 10 machine and the version of cx-freeze is at 7.0.0 (this is what poetry gives me by default). My [pyproject.toml file is here](https://github.com/angelotrivelli/example/blob/main/pyproject.toml). And I have a public repo of this reproducible example here: [https://github.com/angelotrivelli/example/tree/main](https://github.com/angelotrivelli/example/tree/main).

When I delete my venv and downgrade cx-freeze to the version when it was working, 6.15.15, I get a different problem. I can't even install cx-freeze (using `poetry install --no--root`) ...

```
(example-py3.11) PS C:\projects\vscode\example> poetry install --no-root
Updating dependencies
Resolving dependencies... (0.7s)

Package operations: 4 installs, 1 update, 0 removals

  - Installing cx-logging (3.2.0)
  - Installing lief (0.14.0)
  - Updating setuptools (69.0.2 -> 69.5.1)
  - Installing wheel (0.43.0)
  - Installing cx-freeze (6.15.15): Failed

  PermissionError

  [Errno 13] Permission denied: 'C:\\projects\\vscode\\example\\.venv\\Lib\\site-packages\\cx_Freeze\\bases\\console-cpython-311-win_amd64.exe'

  at ~\.pyenv\pyenv-win\versions\3.11.2\Lib\pathlib.py:1044 in open
      1040│         the built-in open() function does.
      1041│         """
      1042│         if "b" not in mode:
      1043│             encoding = io.text_encoding(encoding)
    → 1044│         return io.open(self, mode, buffering, encoding, errors, newline)
      1045│
      1046│     def read_bytes(self):
      1047│         """
      1048│         Open the file in bytes mode, read it, and close the file.

Cannot install cx-freeze.
```

Strangely, it complains about `console-cpython-311-win_amd61` again. I don't know what this file is. 

At this point I am confused. How can I further troubleshoot this?

**Comment by marcelotduarte** [comment]
I imagine it must be a problem with poetry or the use of it.
Several of these [samples](https://github.com/marcelotduarte/cx_Freeze/tree/main/samples) are checked before release and work correctly, including some that use msi.

**Reply by angelotrivelli** [reply]
Thanks Marcelo, 

Just now able to get it working by upgrading to python 3.12, no other changes needed.  ([https://github.com/angelotrivelli/example/tree/with_python_3.12](https://github.com/angelotrivelli/example/tree/with_python_3.12)) 

I will take a close look at setup.py examples to make sure my usage is correct in the future. 

Regards,
-AT.
